force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -122,20 +151,24 @@ public with sharing class ClassesObjectsMethods {
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
         // The variable to store the concatenated string
-
+        String varNames = '';
         // Loop through the list of scientists' names
-
+        for (String currentName : scientistNames){
             // If the name doesn't contain the letter 'G', skip this iteration
-
+            Boolean containsG = (currentName.contains('g') || currentName.contains('G'));
+            
+            if (containsG == true){
```

Comment:
Again, to shorten our code we can combine lines into one more efficient line:
```suggestion 
            if (currentName.containsIgnoreCase('g')){
```
Does that make sense?